### ISSUE TITLE
Language analysis improvements

### ISSUE THREAD

**Issue by ValdisVitolins** [espeak-ng/espeak-ng#199]
Language analysis and spelling decisions could be improved by introducing following new features:
- [ ] extend `verb follows/noun follows` marks to more/arbitrary flags, which then can be used to make different pronunciation rules for homonyms
- [ ] `J` statement as precondition, to allow choosing pronunciation from preceding word
- [ ] possibility to go back to start of the rules and redo analysis again (I.e. not only after removing pre/suf-fixes. Could be performance drain, if used improperly.)
- [ ] `replace` rule extended to replace not only characters, but group of characters, also probably `replace` using matching rules
- [ ] `_list` extended to mark arbitrary defined word types by comparing only root part of the word (i.e. partial match without pre/suffixes)
- [ ] output (prosody data) extended to mark syllables with more/arbitrary defined ways for different pronuciations (e.g. high/low pitch for Chinese etc.) 
- [ ] Fix issue #196 Word end mark _ doesn't work properly with ~ character group.

**Comment by rhdunn** [comment]
Using `.replace` to expand to multiple letters is working for me (i.e. the replace rules in `en_rules`). Are there specific cases that are not working?

**Comment by valdisvi** [comment]
Yes, because in [compiledict.c](https://github.com/espeak-ng/espeak-ng/blob/master/src/libespeak-ng/compiledict.c#L1431) bytes are compressed into integer with [utf8_in](https://github.com/espeak-ng/espeak-ng/blob/master/src/libespeak-ng/translate.c#L572) function, and then only these 4 bytes are  written with [Write4Bytes](https://github.com/espeak-ng/espeak-ng/blob/master/src/libespeak-ng/wavegen.c#L1239). That produces wrong result, if there are too many "meaningful" bytes in _from_ or _to_ part of replacement.
So, universal `.replace` implementation requires to replace arbitrary number of _from_ bytes to arbitrary number of _to_ bytes.
To test it, just add rule e.g.
```
.replace
 æ    are
 are  usi
 ša   ra
//etc. with even more bytes in from or to part of replacement
```
